### ISSUE TITLE
Upgrade to Ruby 2.7

### ISSUE THREAD

**Issue by wvengen** [foodcoops/foodsoft#939]
With Ruby 2.6 recently [gone EOL](https://www.ruby-lang.org/en/downloads/branches/), we better move to 2.7 (or later).
Did a quick check, using 2.7 doesn't break the specs.

Then do a release, finally!

**Comment by wvengen** [comment]
Upgrading to 3.0 is not so easy, unfortunately.
First thing encountered:

```
Failure/Error: value = RailsSettings::CachedSettings["foodcoop.#{self.scope}.#{key}"]

ArgumentError:
  wrong number of arguments (given 1, expected 0)
# ./lib/foodsoft_config.rb:118:in `[]'
# ./lib/foodsoft_config.rb:81:in `block in init_mailing'
# ./lib/foodsoft_config.rb:80:in `each'
# ./lib/foodsoft_config.rb:80:in `init_mailing'
```

Probably RailsSettings::CachedSettings needs a change. We need to replace it anyway, because of #867 (e.g. by just loading all settings on each request, or caching each setting in Redis and keeping it updated in FoodsoftConfig).

**Comment by kidhab** [comment]
I think upgrading to Ruby 2.7 [requires](https://www.fastruby.io/blog/ruby/rails/versions/compatibility-table.html) also an upgrade to Rails 6(.1). Regardless Rails 5.2 is only supported unitl June 2022.

Looking at [RailsBump](https://railsbump.org/lockfiles/a71dc5422c5b62a30db01cc0b789057d55265cca) only [apivore](https://railsbump.org/gems/apivore) isn't supported on newer versions any more. All other gems should be fine. Although there's a [PR](https://github.com/westfieldlabs/apivore/pull/126) adding Rails 6 compability. Or maybe swith to an alternative like [mini-apivore](https://github.com/alekseyl/mini-apivore).

**Comment by Viehlieb** [comment]
Hey folks,
We are currently working on the **ruby upgrade** and plan to upgrade to 6.1 anyways. 
We got some funding for further developing foodsoft. Check out the project website or if you want to get in touch about the upgrade, contact me or the team:

https://prototypefund.de/project/weiterentwicklung-von-foodsoft/ 

We'll inform you about our process.

For **Rails Upgrade** we created a separate issue and we are currently working on that.
https://github.com/foodcoops/foodsoft/issues/956

Best regards
